core/src/main/scala/kafka/admin/AdminUtils.scala

```diff
@@ -316,17 +323,16 @@ object AdminUtils extends Logging with AdminUtilities {
     var partitionId = startPartitionId
     partitionList = partitionList.takeRight(partitionList.size - partitionId)
     for (i <- partitionList.indices) {
-      val brokerList = partitionList(i).split(":").map(s => s.trim().toInt)
+      val brokerList = partitionList(i).split(":").map(s => s.trim().toInt).toList
       if (brokerList.isEmpty)
-        throw new AdminOperationException("replication factor must be larger than 0")
+        throw new InvalidReplicaAssignmentException("Replication factor must be larger than 0.")
```

Comment:
We should include the partition id in the message. Same for every other error message.